src/Liquid/AbstractBlock.php

```diff
@@ -51,56 +51,56 @@ public function getNodelist()
 	 */
 	public function parse(array &$tokens)
 	{
-		$startRegexp = new Regexp('/^' . Liquid::get('TAG_START') . '/');
-		$tagRegexp = new Regexp('/^' . Liquid::get('TAG_START') . Liquid::get('WHITESPACE_CONTROL') . '?\s*(\w+)\s*(.*?)' . Liquid::get('WHITESPACE_CONTROL') . '?' . Liquid::get('TAG_END') . '$/s');
-		$variableStartRegexp = new Regexp('/^' . Liquid::get('VARIABLE_START') . '/');
-
 		$this->nodelist = array();
 
 		$tags = Template::getTags();
 
-		while (count($tokens)) {
-			$token = array_shift($tokens);
+		for ($i = 0, $n = count($tokens); $i < $n; $i++) {
+			if ($tokens[$i] === null) {
+				continue;
+			}
+			$token = $tokens[$i];
+			$tokens[$i] = null;
+
+			if (preg_match('/({%)?(-)?(?(1)\s*(\w+)|({{)(-)?)\s*(.*?)\s*(-)?([%}]})/s', $token, $tokenParts)) {
```

Comment:
How can we explain that the assumption that someone might override `TAG_START` is wrong?

Reply:
I could "cached" this in the config array with calls to `TAG_START` et al.

Also, why are these configurable? What conceivable situation requires different tag/variable bracket syntax? This seems over-engineered.

Reply:
I agree with the last point wholeheartedly, yet here we are.

Reply:
Then those options could be dropped in the future in a breaking release (v2.0.0)

I have modified the regular expression to allow for those configuration options.